lib/spack/spack/package.py

```diff
@@ -525,6 +525,7 @@ def extendee_spec(self):
         name = next(iter(self.extendees))
 
         # If the extendee is in the spec's deps already, return that.
+        # XXX(deptype): DEP_?
```

Comment:
probably only want to search link and run deps here.  I don't think you'd ever extend something that was only a build dep -- in fact, I think it should be an error to have a build-only extension relationship.  The extension concept assumes the extendee will stick around for the lifetime of the extension -- the extension has to be linked/unlinked from the directory of the extendee.